src/sentry/monitors/endpoints/monitor_ingest_checkin_index.py

```diff
@@ -145,6 +145,7 @@ def post(
                         "config": monitor_data["config"],
                     },
                 )
+                signal_first_monitor_created(project, request.user, True)
```

Comment:
i noticed we are using update_or_create for possible race condition, is it advised to do something similar here to not double count upserts?

Reply:
`update_or_create` returns a bool as the second param to say whether we created or not, we could skip firing this here. 

Might also make sense to do https://github.com/getsentry/sentry/blob/044befcb322457033b767e40a156c56458909b38/src/sentry/rules/processor.py#L212-L219

Basically only update the flag if it's not already set, and skip firing the event if we don't update any rows

Reply:
oooh very nice, will implement both of these guys